lib/src/client_io_offline.dart

```diff
@@ -0,0 +1,381 @@
+import 'dart:async';
```

Comment:
Implementation wise, we don't need this separate `client_io_offline`, we can modify the call method of  `client_io` as required, and keep all the offline related functions that are not exposed to `Client` interface in the `client_offline_mixin`

Reply:
@lohanidamodar the problem I had was with some of the members in the class vs mixin. For example, `processWriteQueue()` is an offline-specific method, but it needs access to `call()` defined in `Client` rather than the mixin. Any suggestions on how to handle this?

Reply:
we can may be provide callback?

Reply:
I've moved as much of the offline logic as I could into a mixin, updated the ClientIO class to use the mixin and deleted the subclass. Let me know what you think.